src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -104,6 +113,54 @@ public Post updatePost(Long courseId, Post post) {
         return updatedPost;
     }
 
+    /**
+     * Checks course, user and post validity,
+     * updates the pin state, persists the post,
+     * and ensures that sensitive information is filtered out
+     *
+     * @param courseId  id of the course the post belongs to
+     * @param postId    id of the post to change the pin state for
+     * @param pinState  new boolean value of the pinned flag for the given post
+     * @return updated post that was persisted
+     */
+    public Post updatePinState(Long courseId, Long postId, Boolean pinState) {
```

Comment:
Maybe consider doing this check as part of `Post::setPinned`? I would guess I always want to keep this consistent?